torch/csrc/jit/codegen/cuda/lower_index.cpp

```diff
@@ -148,6 +146,24 @@ void IndexLowering::handle(const ARangeOp* aop) {
   GpuLower::current()->propagateExprInfo(aop, back());
 }
 
+void IndexLowering::handle(const EyeOp* eop) {
+  auto out_tv = dynamic_cast<TensorView*>(eop->output(0));
+  TORCH_INTERNAL_ASSERT(out_tv != nullptr);
+
+  // linear index for computing eye output
+  auto indices = Index::getPerDimRootIndices(out_tv, for_loops_);
+  TORCH_INTERNAL_ASSERT(indices.size() == 2);
+  auto index1 = indices[0];
```

Comment:
Just checked with https://pytorch.org/docs/stable/generated/torch.eye.html, and there is no batching dimension, but they do have an `m` parameter for generating non-square matrices.  I will add that. Regarding the `TensorIndex` question, I think because these two `Val`s are all we need, `TensorIndex` is an overkill because we don't need any other functionalities that `TensorIndex` has.

Reply:
My feeling for `TensorIndex` vs `Val` is: if you want a string like `T1[index]`, then you should use `TensorIndex`, otherwise, you just need a `Val`. Is this understanding correct?

Reply:
I think so, and was just a curious question. 

I guess either way makes sense in here as we were actually "indexing" an "eye" tensor that we happened to have inlined away.